docs/architecture/deep-dives/authentication/index.md

```diff
@@ -0,0 +1,325 @@
+    the user will also have custom claims added as defined for that org.
+- **Provider claims**
+  - `providerprovideradmin` - The `ID` of the provider(s) for which the user is an admin.
+  - `providerserviceuser` - The `ID` of the provider(s) for which the user is a user.
+
+For API token requests (grant type of `client_credentials`), the response differs based on the
+client type, as follows:
+
+<!-- prettier-ignore -->
+| Client Type  | Claims |
+| ------------ | ------ |
+| Installation | `sub` - the installation `ID` |
+| Internal     | `sub` - the internal `ID` |
+| Organization | `sub` - the organization `ID` |
+| User         | `sub` - the user ID <br /> `amr` - set to `Application` and `external` <br /> Plus all of the user claims from a password authentication request as listed above. |
+
+#### Decryption Information
+
+The response will also contain a custom JSON response object containing the following properties:
+
+<!-- prettier-ignore -->
+| Property | Purpose |
+| -------- | -------- |
+| `PrivateKey` | The user’s RSA private key, encrypted with the user's symmetric encryption key. |
+| `Key` | The user’s symmetric encryption key, encrypted with the user's master password. |
+| `UserDecryptionOptions` | The ways that the user has available to them to decrypt their symmetric key. See [User Decryption Options](#appendix-userdecryptionoptions-details) for details. |
+| `Kdf` | The user’s defined KDF type. |
+| `KdfIterations` | The user’s defined KDF iterations. |
+| `ForcePasswordReset`  | A boolean indicating whether the user must immediately reset their password. |
+| `MasterPasswordPolicy` | The master password policy for any organizations of which the user is a member. |
+| `TwoFactorToken` | A token that can be used to bypass 2FA. Generated when a user chooses to "remember" their 2FA response. |
+| `ResetMasterPassword` (_obsolete_) | A boolean indicating whether the user needs to set their master password. It is explicitly set to `false` for users with Key Connector configured. This has been superseded by `UserDecryptionOptions.HasMasterPassword` but has been left for backward compatibility. |
+
+For API token requests, the following additional properties are added to the response:
+
+- `ApiUseKeyConnector` - set to `true` if the user is set to use Key Connector for master password
+  retrieval.
+
+## Step 3: Client handles the response
+
+When a token request is verified and sent back to the client, the client is responsible for storing
+the resulting data in state properly, so that the user can authenticate and their vault can be
+decrypted.
+
+The core logic for handling the token responses is in the `processTokenResponse()` method in our
+base
+[`LoginStrategy`](https://github.com/bitwarden/clients/blob/master/libs/common/src/auth/login-strategies/login.strategy.ts).
+There, we do the following:
+
+### Use the Authentication Information to Set up the User’s Account
```

Comment:
⛏️ Shouldn't need the running capitalization.

Reply:
Adjusted this to flow without capitalization.